src/File.jl

```diff
@@ -31,45 +32,53 @@ File:
 struct File
     path::String
     # This is the metadata from JSON-sidecar (doesn't exists in BIDS-like directory)
-    metadata::OrderedDict{String, Any}
+    metadata::OrderedDict{String,Any}
     # Entities are the key-value from filename
-    entities::OrderedDict{String, String}
+    entities::OrderedDict{String,String}
+    events::DataFrame
 end
 
 function File(
     path::AbstractString;
     load_metadata::Bool=true,
     require_modality::Bool=true,
     strict::Bool=true,
-    extract_from_full_path::Bool=true
+    extract_from_full_path::Bool=true,
+    load_events::Bool=true
 )
     entities = parse_path(path, require_modality=require_modality, strict=strict)
     # Try extracting sub and ses from full path if not exists in parsed filename
     if extract_from_full_path
         if !haskey(entities, "sub")
             entities["sub"] = get_sub(
-                                path,
-                                from_fname=false,
-                                require_modality=require_modality,
-                                strict=strict
-                              )
+                path,
+                from_fname=false,
+                require_modality=require_modality,
+                strict=strict
+            )
         end
         if !haskey(entities, "ses")
             entities["ses"] = get_ses(
-                                path,
-                                from_fname=false,
-                                require_modality=require_modality,
-                                strict=strict
-                              )
+                path,
+                from_fname=false,
+                require_modality=require_modality,
+                strict=strict
+            )
         end
     end
     metadata = !load_metadata ? OrderedDict{String,Any}() :
                JSON.parsefile(
-                   get_metadata_path(path), dicttype=OrderedDict{String,Any}
-               )
-    File(path, metadata, entities)
+        get_metadata_path(path), dicttype=OrderedDict{String,Any}
+    )
+    events = !load_events ? DataFrame() : CSV.read(
+        get_events_path(path),
```

Comment:
If I understand correctly, `get_events_path(path)` could return `nothing` if for some reason the corresponding `_events.tsv` doesn't exist. 

If that's the case, the `CSV.read(nothing)` will raise `MethodError: no method matching joinpath(::Nothing)`, which could be unhelpful for user. 

I suggest raising a more useful error in `get_events_path(path)` that indicates `joinpath(this_dirname, fname_without_ext * "_events.tsv")` couldn't be found.